pyfstat/grid_based_searches.py

```diff
@@ -750,22 +763,22 @@ def run(self, return_data=False):
             return
 
         if hasattr(self, "search") is False:
-            self.inititate_search_object()
+            self._initiate_search_object()
 
-        data = []
-        if self.outputTransientFstatMap:
-            self.tCWfilebase = os.path.splitext(self.out_file)[0] + "_tCW_"
-            logging.info(
-                "Will save per-Doppler Fstatmap"
-                " results to {}*.dat".format(self.tCWfilebase)
-            )
+        output_dtype = np.dtype(
+            {
+                "names": self.output_keys,
+                "formats": np.repeat(float, len(self.output_keys)),
+            }
+        )
+        data = np.zeros(len(self.input_data), dtype=output_dtype)
```

Comment:
Same as before, what?

Reply:
Same logic as before, but on rereading this, I'm wondering why in this case they're still all `float` and no `int` for the `["t0", "tau"]` parameters... Hmm...

Reply:
I'm not familiar with these parameters.

We use `int` for time whenever time is set using some variation of `LALEpoch` or `LALGPS` or that sort of thing (otherwise parsing becomes a nightmare, as seen multiple times in this package). 

Are these parameters treated as epochs in LALSuite?

Reply:
...[no](https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/structtransient_window_range__t.html)...

Reply:
I see, they are literally what in python would be `int` (if my understanding about swig magic is not mistaken).

Reply:
...while [CFSv2](https://git.ligo.org/lscsoft/lalsuite/-/blob/master/lalapps/src/pulsar/Fstatistic/ComputeFstatistic_v2.c#L1086) takes either `transient_t0Epoch`, which is an `EPOCH` type, or `transient_t0Offset`, which is not, and that makes sense. `tau` (duration) by definition is always a length, so doesn't need to be an epoch.

Reply:
```suggestion
        output_dtype = np.dtype(
            {
                "names": self.output_keys,
                "formats": np.array([float if key not in ["t0", "tau"] else int for key in self.output_keys]),
            }
        )
        data = np.zeros(len(self.input_data), dtype=output_dtype)
```

Reply:
This does work, but has one problem: if an existing data file is found, `np.genfromtxt` (called via `helper_functions.read_txt_file_with_header`) guesses the dtypes as float (even if in the text file none of the t0/tau colum entries have any decimals, as our _output_ format for those is already `%d`). So in that case, `self.data = old_data` will have all-float dtype, while if an actual search is run, it will have `[float,float,...,int,int]`. If one just runs a search and writes the output to disk, there's no problem, as the mentioned output formatting will make the difference disappear. But if someone works with the search class interactively, this might cause some confusion.

Reply:
We decided for now to leave it as is: no real problems from having them all stored as float internally. The whole thing can always get improved further in the future, but then we may want to switch to a unified dict-based storage for both grid- and MCMC-searches anyway, making extra work on this superfluous for now.